tests/integration/__init__.py

```diff
@@ -54,3 +58,30 @@ def set_up_record(self):
                     user_agent=pytest.placeholders.user_agent,
                     refresh_token=pytest.placeholders.refresh_token,
                 )
+
+    def use_cassette(self, cassette_name=None, **kwargs):
+        """Use a cassette. The cassette name is dynamically generated.
+
+        :param cassette_name: (Deprecated) The name to use for the cassette. All names
+            that are not equal to the dynamically generated name will be logged.
+        :param kwargs: All keyword arguments for the main function
+            (``Betamax.use_cassette``).
+        """
+        dynamic_name = self.get_cassette_name()
+        if cassette_name:
+            self.logger.debug(
+                "Static cassette name provided by {}. The following "
+                "name was provided: {}".format(dynamic_name, cassette_name)
+            )
+            if cassette_name != dynamic_name:
+                self.logger.warning(
+                    "Dynamic cassette name for function {} does not "
+                    "match the provided cassette name: {}".format(
+                        dynamic_name, cassette_name
+                    )
+                )
+        return self.recorder.use_cassette(cassette_name or dynamic_name, **kwargs)
+
+    def get_cassette_name(self) -> str:
+        function_name = inspect.currentframe().f_back.f_back.f_code.co_name
```

Comment:
`inspect` isn't needed to get the current test name as pytest will set `PYTEST_CURRENT_TEST ` for each test.
```suggestion
        current_test = os.environ.get('PYTEST_CURRENT_TEST')
        _, test_name = currentTest.split('::')
```

Reply:
I prefer my method because it doesn't rely on parsing an environment variable, of which the formatting is subject to change in any major version.

Reply:
@bboe @jarhill0 Can you leave your opinion on which method should be used?

Reply:
> formatting is subject to change in any major version
Has is changed significantly before? If so, we can easily update that.

Reply:
At this point it's a matter of preference rather than readability or function.

Reply:
Functionally, using inspect is faster than your method.

Here's the implementation of my code:

```python
class TestClass:
    def get_cassette_name(self) -> str:
        function_name = inspect.currentframe().f_back.f_back.f_code.co_name
        return "{}.{}".format(type(self).__name__, function_name)
    def use_cassette(self):
        name = self.get_cassette_name()
        return name
    def test_something(self):
        assert self.use_cassette() == "TestClass.test_something"  
```

```python
In[14]: %timeit tc.test_something()
1.18 µs ± 7.03 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

And here is the implementation of your example:

```python
class TestClass:
    def get_cassette_name(self) -> str:
        current_test = os.environ.get('PYTEST_CURRENT_TEST')
        class_name, test_name = current_test.split('::')
        return "{}.{}".format(class_name, test_name)
    def use_cassette(self):
        name = self.get_cassette_name()
        return name
    def test_something(self):
        assert self.use_cassette() == "TestClass.test_something"  
```

```python
In[33]: os.environ["PYTEST_CURRENT_TEST"] = "TestClass::test_something"
In[34]: %timeit tc.test_something()
2.28 µs ± 10.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

As you can see, my example is almost twice as fast.

Reply:
Functionally, I don't think 1.10 µs is significant enough to actually matter.

Reply:
So then it comes right back down to preference.

Reply:
> @jarhill0 Can you leave your opinion on which method should be used?

Doesn't matter to me. They seem equally reliable.

Reply:
Personally, examining the stack is adding complexity that doesn't need to be there.

Reply:
I do agree that stack and code types have a lot of variables with names that don’t make too much sense, but I would consider it more Pythonic.

Reply:
How so? How often do you see packages utilize `inspect`?

Reply:
But inspect is standard library, which gives it more reliability than third party packages.

Reply:
My method does not use any new 3rd party packages.

Reply:
I don't particularly care either way as this is a form of bikeshedding. However, here's my 2c since I was asked.

I personally like the simplicity of the environment variable approach. Unfortunately, pytest has this warning:

> The contents of PYTEST_CURRENT_TEST is meant to be human readable and the actual format can be changed between releases (even bug fixes) so it shouldn’t be relied on for scripting or automation.

Since we are precisely relying on it for scripting, I thus would avoid that method, in which case using inspect appears to be the way to go. And, since inspect is a standard library I'm fine with that.

Reply:
I was not aware of that warning. In that case, lets do the inspect method.